src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -203,6 +224,112 @@ public override void Return(T[] array, bool clearArray = false)
             }
         }
 
+        public bool Trim()
+        {
+            uint tickCount = (uint)Environment.TickCount;
+
+            MemoryPressure pressure = GetMemoryPressure();
+            foreach (PerCoreLockedStacks bucket in _buckets)
+            {
+                bucket?.Trim(tickCount, Id, pressure);
+            }
+
+            ArrayPoolEventSource log = ArrayPoolEventSource.Log;
+            uint trimTicks = pressure == MemoryPressure.High ? TlsHighTrimAfterTicks : TlsTrimAfterTicks;
+            foreach (var tlsBuckets in s_allTlsBuckets)
+            {
+                Bucket[] buckets = tlsBuckets.Key;
+                for (int i = 0; i < NumBuckets; i++)
+                {
+                    uint stockTicks = buckets[i].Ticks;
+                    if (stockTicks > tickCount || (tickCount - stockTicks) > trimTicks)
+                    {
+                        T[] buffer = buckets[i].Buffer;
+
+                        // We've wrapped the tick count or elapsed enough time since the
+                        // first item went into this thread local bucket. Clear it.
+                        buckets[i].Buffer = null;
+
+                        if (log.IsEnabled() && buffer != null)
+                        {
+                            log.BufferTrimmed(buffer.GetHashCode(), buffer.Length, Id);
+                        }
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// This is the static function that is called from the gen2 GC callback.
+        /// The input object is the instance we want the callback on.
+        /// </summary>
+        /// <remarks>
+        /// The reason that we make this function static and take the instance as a parameter is that
+        /// we would otherwise root the instance to the Gen2GcCallback object, leaking the instance even when
+        /// the application no longer needs it.
+        /// </remarks>
+        private static bool Gen2GcCallbackFunc(object target)
+        {
+            return ((TlsOverPerCoreLockedStacksArrayPool<T>)(target)).Trim();
+        }
+
+        private enum MemoryPressure
+        {
+            Low,
+            Medium,
+            High
+        }
+
+        private static MemoryPressure GetMemoryPressure()
+        {
+            GC.GetMemoryInfo(out uint threshold, out _, out uint lastLoad, out _, out _);
+            if (lastLoad >= threshold * .90)
+            {
+                return MemoryPressure.High;
+            }
+            else if (lastLoad >= threshold * .80)
+            {
+                return MemoryPressure.Medium;
+            }
+            return MemoryPressure.Low;
+        }
+
+        private static bool TrimBuffers { get; } = GetTrimBuffers();
+
+        private static bool GetTrimBuffers()
```

Comment:
Do you need both a property and method that do the same thing

Reply:
If I want it cached, yes.

Reply:
They're not actually the same thing... TrimBuffers is backed by a static readonly that's being initialized to the result of GetTrimBuffers.